wordpress-mobile/WordPress-iOS
# [Domain Focus] Card Logic

This PR is a follow up for [PR 21241](https://github.com/wordpress-mobile/WordPress-iOS/pull/21241). It implements the logic for the card along with tracks.

![Simulator Screenshot - iPhone 14 Pro - 2023-08-08 at 12 40 02](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/8dc0a2cb-1d78-459e-8792-f2cf0f6b1539)

## Testing:

1. Install & Launch Jetpack
2. Enable the feature flag.
3. Navigate to MySite (either kill the app first or open a different tab and come back to My Site).
4. Verify the card is being shown.
5. Verify the track `dashboard_card_domain_transfer_shown` is fired.

### Button & Card
- Verify tapping on the button & the card opens the [correct url](https://wordpress.com/transfer-google-domains/) in webview.
- Verify the track `dashboard_card_domain_transfer_button_tapped` is fired.

### More Button
- Verify if the more button opens a menu that contains the hide functionality.
- Verify the track `dashboard_card_domain_transfer_more_menu_tapped` is fired.

### Hide
- Tap one more & hide the card.
- Verify the default hidden card track is fired for Domain Focus Card.

### WordPress App
- The card shouldn't be visible on the WordPress app.

## Regression Notes
1. Potential unintended areas of impact
The card could appear in the wrong order. 

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tested.

7. What automated tests I added (or what prevented me from doing so)
It is a predominantly UI addition along with tracks. I didn't add unit tests.

PR submission checklist:
